components/ArticleCard.js

```diff
@@ -1,14 +1,106 @@
 import styled from "styled-components";
-export default function ArticleCard({ articles, onArticleAddSubmit }) {
+import { useState } from "react";
+
+export default function ArticleCard({
+  articles,
+  onArticleAddSubmit,
+  setArticles,
+}) {
+  const [articleEdit, setArticleEdit] = useState(null);
+  const [editingArticle, setEditingArticle] = useState({
+    articleName: "",
+    price: "",
+  });
+
+  const handleArticleEditClick = (event, article) => {
+    event.preventDefault();
+    setArticleEdit(article.articleNumber);
+
+    const formValues = {
+      articleName: article.articleName,
+      price: article.price,
+    };
+    setEditingArticle(formValues);
+  };
+
+  const handleEditArticleFormChange = (event) => {
+    event.preventDefault();
+
+    const fieldName = event.target.name;
+    const fieldValue = event.target.value;
+
+    const newFormArticleData = { ...editingArticle };
+    newFormArticleData[fieldName] = fieldValue;
+    setEditingArticle(newFormArticleData);
+  };
+
+  const handleEditFormSubmit = (event) => {
+    event.preventDefault();
+
+    const editedArticle = {
+      articleNumber: articleEdit,
+      articleName: editingArticle.articleName,
+      price: +editingArticle.price,
+    };
+
+    const newArticles = [...articles];
+
+    const index = articles.findIndex(
+      (article) => article.articleNumber === articleEdit
+    );
+    newArticles[index] = editedArticle;
+
+    setArticles(newArticles);
+    setArticleEdit();
+  };
+
   return (
     <>
       {articles.map((article) => (
         <StyledUl key={article.articleNumber}>
-          <li>
-            <p>{article.articleNumber}</p>
-            <p>{article.articleName}</p>
-            <p>{article.price} €</p>
-          </li>
+          {articleEdit === article.articleNumber ? (
+            <>
+              <li>
+                <p>{article.articleNumber}</p>
+                <p>{article.articleName}</p>
+                <p>{article.price} €</p>
+              </li>
+              <li>
+                <StyledForm onSubmit={handleEditFormSubmit}>
+                  <label htmlFor="articleName">Articlename</label>
+                  <input
+                    type="text"
+                    id="articleName"
+                    name="articleName"
+                    value={editingArticle.articleName}
```

Comment:
wenn du ihm hier schon ein Value-Attribut mitgibst, kann das dann noch geändert werden? wäre das in dem Fall vielleicht eher ein defaultValue?

Reply:
das funktioniert einwandfrei